src/Categories/Object/NaturalNumber.agda

```diff
@@ -0,0 +1,59 @@
+{-# OPTIONS --without-K --safe #-}
+
+open import Categories.Category.Core
+open import Categories.Object.Terminal hiding (up-to-iso)
+
+module Categories.Object.NaturalNumber {o ℓ e} (𝒞 : Category o ℓ e) (𝒞-Terminal : Terminal 𝒞) where
+
+open import Level
+
+open import Categories.Morphism 𝒞
+open import Categories.Morphism.Reasoning 𝒞
+
+open Category 𝒞
+open HomReasoning
+open Equiv
+
+open Terminal 𝒞-Terminal
+
+private
+  variable
+    A B C D X Y Z : Obj
+    h i j : A ⇒ B
+
+record NaturalNumber : Set (o ⊔ ℓ ⊔ e) where
```

Comment:
not sure if would make sense to provide a predicate version, like `IsNaturalNumber`?

Reply:
Good point! We should probably do this for all of our `Object`s moving forward 🙂 Perhaps we should add something to the design doc about this?

Reply:
let's put some thought into it. in fact, all definitions of objects and diagrams are in predicate style.

Reply:
I can think of a few off the top of my head that aren't, but definitely should be (`Pullback` and `Coequalizer` are ones that comes to mind immediately)

Reply:
yeah probably `Terminal` is not. `Product` is though. sort of inconsistent indeed.